### PR TITLE
fix: CallKit integration FS-1142

### PR DESCRIPTION
**Pull request by johnxnguyen**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-1142" title="FS-1142" target="_blank">TaskFS-1142</a>  [iOS] CallKit not working
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

This PR includes a variety of changes to help resolve issues with CallKit integration:
- add helpers to determine if a call event should start or stop the ringing sound
- simplify how we share known calls with the notification service extension

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Sources/Request Strategies/Feature configurations/GetFeatureConfigsActionHandlerTests.swift [hunk 1]
```diff
@@ -59,44 +59,48 @@ class GetFeatureConfigsActionHandlerTests: MessagingTestBase {
     // MARK: - Response handling
 
     func test_ItHandlesResponse_200() throws {
-        // Given
-        let sut = GetFeatureConfigsActionHandler(context: syncMOC)
-        var action = GetFeatureConfigsAction()
+        syncMOC.performGroupedBlock {
```
Comment (johnxnguyen) [inline Sources/Request Strategies/Feature configurations/GetFeatureConfigsActionHandlerTests.swift:62]: This isn't strictly related to the PR, but it was a test that was failing for some reason and wrapping it in a `syncMOC.performGroupedBlock` fixed it.

**Review by KaterinaWire — APPROVED** [review]

**Comment by johnxnguyen** [comment]
release major

**Comment by zenkins** [comment]
Hello, @johnxnguyen! Merging PR for command **release major** for PR #**582**. Will create a **major** release after merged.
Next version is 273.0.0.

**Comment by johnxnguyen** [comment]
release major

**Comment by zenkins** [comment]
Hello, @johnxnguyen! Merging PR for command **release major** for PR #**582**. Will create a **major** release after merged.
Next version is 273.0.0.

**Pull request merged**